src/respend/respendrelayer.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2018 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "respend/respendrelayer.h"
+#include "net.h" // RelayTransaction
+#include "primitives/transaction.h"
+#include "protocol.h"
+#include "streams.h"
+#include "util.h"
+#include "utilprocessmsg.h"
+#include <mutex>
+
+namespace respend {
+
+namespace {
+
+// Apply an independent rate limit to double-spend relays
+class RelayLimiter {
+    public:
+        RelayLimiter() : respendCount(0), lastRespendTime(0) { }
+
+        bool HasLimitExceeded(const CTransaction& doubleSpend)
+        {
+            unsigned int size = ::GetSerializeSize(doubleSpend,
+                                                   SER_NETWORK, PROTOCOL_VERSION);
+
+            std::lock_guard<std::mutex> lock(cs);
+            int64_t limit = GetArg("-limitrespendrelay", 100);
+            if (RateLimitExceeded(respendCount, lastRespendTime, limit, size)) {
```

Comment:
use DEFAULT_LIMITRESPENDRELAY in place of the 100.

Reply:
done in a later commit